client/clientservice/src/state_snapshot_service.cpp

```diff
@@ -202,19 +202,23 @@ static std::shared_ptr<bftEngine::RequestCallBack> getCallbackLambda(const bftEn
   return std::make_shared<bftEngine::RequestCallBack>(callback);
 }
 
-Status StateSnapshotServiceImpl::GetRecentSnapshot(ServerContext* context,
-                                                   const GetRecentSnapshotRequest* proto_request,
-                                                   GetRecentSnapshotResponse* response) {
-  std::chrono::milliseconds timeout = 5s;  // This is default timeout
+void StateSnapshotServiceImpl::setTimeoutFromDeadline(ServerContext* context, std::chrono::milliseconds& timeout) {
```

Comment:
Nit: I would leave the default timeout setting in the scope of this function and not set it from outside:
std::chrono::milliseconds StateSnapshotServiceImpl::setTimeoutFromDeadline(ServerContext* context) {
  std::chrono::milliseconds timeout = 30s;
  .....
  return timeout;
}
...
Caller:
std::chrono::milliseconds  timeout = setTimeoutFromDeadline(...);

Reply:
Thanks, @yuliasherman will update the PR with this. This is really nice.